### DISCUSSION TITLE
Operator handling of Kube API failures

### DISCUSSION THREAD

**Discussion by rareddy** [strimzi/strimzi-kafka-operator#5573] [Q&A]
Once every couple of days we see errors like below in Strimzi log and Operator turns the status of the Kafka resource to "Error", and in the next reconciliation cycle the error is corrected and the system comes back to normal. I suspect the issue is probably with `kube-apiserver` not honoring requests made for the resource ( we have seen resource type differ each time as it could be PVC, StatefulSet, Secret etc). This, in turn, generates some false alerts for us.

my question is there a case to be made to be flexible in retrying the resource before flipping the Status to an Error? for example, fabric8 Kube client API provides a property like `kubernetes.request.retry.backoffLimit` to define retry count before status is changed?

```
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://172.30.0.1/api/v1/namespaces/mk-c4jr0s5o6rpibn45kdu0/persistentvolumeclaims?labelSelector=strimzi.io%2Fcluster%3Dnubium-transformations-dev%2Cstrimzi.io%2Fname%3Dnubium-transformations-dev-zookeeper%2Cstrimzi.io%2Fkind%3DKafka. 
Message: rpc error: code = Unavailable desc = transport is closing. 
Received status: Status(apiVersion=v1, code=500, details=null, kind=Status, message=rpc error: code = Unavailable desc = transport is closing, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=null, status=Failure, additionalProperties={}).
```

**Comment by shawkins** [comment]
Unfortunately the fabric8 retry is currently only for network / IOExceptions.  5xx errors would require an enhancement to issue use the retry logic.

**Reply by shawkins** [reply]
Sorry I misread the changes, yes this situation is handled by https://github.com/fabric8io/kubernetes-client/blob/7c31eb13204570f1255d96ab7d216064f034872e/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java#L592 - so setting a backoff limit should help with this.
